dan_layer/state_store_sqlite/src/reader.rs

```diff
@@ -720,6 +720,36 @@ impl<'tx, TAddr: NodeAddressable + Serialize + DeserializeOwned + 'tx> StateStor
         block.try_convert(qc)
     }
 
+    fn blocks_get_first_in_epoch(&self, epoch: Epoch) -> Result<Option<Block>, StorageError> {
```

Comment:
We typically use the [optional combinator](https://github.com/tari-project/tari-dan/blob/development/dan_layer/common_types/src/optional.rs) instead of returning Option<T>

e.g.
```
// .optional() changes Result<T, StorageError> to Result<Option<T>, StorageError> if a NotFound error is returned
tx.blocks_get_first_in_epoch(epoch).optional()?
```

This both simplifies the implementation a bit and allows the caller to choose if they expect a NotFound error 

```rust
 let row = blocks::table
            .left_join(quorum_certificates::table.on(blocks::qc_id.eq(quorum_certificates::qc_id)))
            .select((blocks::all_columns, quorum_certificates::all_columns.nullable()))
            .filter(blocks::epoch.eq(epoch.as_u64() as i64))
            .filter(blocks::height.eq(0))
            .first::<(sql_models::Block, Option<sql_models::QuorumCertificate>)>(self.connection())
            .map_err(|e| SqliteStorageError::DieselError {
                operation: "blocks_get_first_in_epoch",
                source: e,
            })?;

            let qc = qc.ok_or_else(|| SqliteStorageError::DbInconsistency {
                operation: "blocks_get_first_in_epoch",
                details: format!(
                    "block {} references non-existent quorum certificate {}",
                    block.id, block.qc_id
                ),
            })?;
            let res = block.try_convert(qc)?;
            Ok(res)
```

Reply:
We could also name this `blocks_get_genesis_for_epoch`

Reply:
It is possible that the VN is not registered for the next epoch or doesnt have blocks for the previous epoch(s). We may have to request historical transaction receipts rather than blocks in future.